### PR TITLE
Upgrade the dang thing

### PR DESCRIPTION
**Pull request by jherdman**
It's time to support Rails 8

### CODE REVIEW

File: nulogy-fitter-happier.gemspec [hunk 1]
```diff
@@ -25,12 +25,12 @@ Gem::Specification.new do |s|
   s.test_files   = Dir["spec/**/*"]
   s.require_path = 'lib'
 
-  s.add_dependency 'actionpack', '>= 6.0', '< 8.0'
-  s.add_dependency 'activerecord', '>= 6.0', '< 8.0'
+  s.add_dependency 'actionpack', '>= 7.2', '< 8.0'
+  s.add_dependency 'activerecord', '>= 7.2', '< 8.0'
   s.add_dependency 'newrelic_rpm', '>= 6.15.0', '< 10.0'
   s.add_dependency 'railties', '>= 6.0', '< 8.0'
```
Comment (amckinnell) [inline nulogy-fitter-happier.gemspec]: The version of `railties` should match that of `actionpack` and `activerecord` as they are always released together and match a particular version of the `rails` gem.

File: nulogy-fitter-happier.gemspec [hunk 2]
```diff
@@ -25,12 +25,12 @@ Gem::Specification.new do |s|
   s.test_files   = Dir["spec/**/*"]
   s.require_path = 'lib'
 
-  s.add_dependency 'actionpack', '>= 6.0', '< 8.0'
-  s.add_dependency 'activerecord', '>= 6.0', '< 8.0'
+  s.add_dependency 'actionpack', '>= 7.2', '< 8.0'
```
Comment (amckinnell) [inline nulogy-fitter-happier.gemspec]: I would prefer that we specify `'>= 7.0'` as we are still using this gem with earlier versions of `Rails` (in particular for `DQI` and `PackManager`).

**Comment by amckinnell** [comment]
The version needs to be updated in the `gemspec` and the `CHANGELOG` needs to be updated.

File: .ruby-version [hunk 3]
```diff
@@ -1 +1 @@
-2.7.6
+3.4.1
```
Comment (amckinnell) [inline .ruby-version:1]: I usually stay a little behind the latest ruby (especially in this case as `3.4` was released so recently). I would suggest the latest of the `3.3.x` versions. For this particular gem it probably doesn't matter as I doubt that there is benefit to using ruby `3.4.x` features so probably no one will.

File: nulogy-fitter-happier.gemspec (hunk 1 above)
Comment (jherdman) [reply nulogy-fitter-happier.gemspec]: smh

**Comment by jherdman** [comment]
Closed in favour of #11

**Pull request closed without merging**